arcgis-ios-sdk-samples/Scenes/Choose camera controller/ChooseCameraControllerViewController.swift

```diff
@@ -0,0 +1,184 @@
+// Copyright 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ChooseCameraControllerViewController: UIViewController {
+    
+    @IBOutlet private var sceneView: AGSSceneView!
+    @IBOutlet private var tableView: UITableView!
+    @IBOutlet private var visualEffectView: UIVisualEffectView!
+    
+    var cameraControllers = [AGSCameraController]()
+    let longitude = -109.937516, latitude = 38.456714
+    
+    lazy var planeSymbol: AGSModelSceneSymbol = {[unowned self] in
+        let planeSymbol = AGSModelSceneSymbol(name: "Bristol", extension: "dae", scale: 100.0)
+        planeSymbol.load{ _ in
+            DispatchQueue.main.async {
+                self.planeSymbolDidLoad()
+            }
+        }
+        return planeSymbol
+    }()
+    
+    lazy var planeGraphic: AGSGraphic = {
+        let planePosition = AGSPoint(x: longitude, y: latitude, z: 5000, spatialReference: AGSSpatialReference.wgs84())
+        let planeGraphic = AGSGraphic(geometry: planePosition, symbol: planeSymbol, attributes: nil)
+        return planeGraphic
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["ChooseCameraControllerViewController"]
+        
+        // Constraint visual effect view to the scene view's attribution label.
+        visualEffectView.bottomAnchor.constraint(equalTo: sceneView.attributionTopAnchor, constant: -10).isActive = true
+        
+        // Assign the scene to the scene view.
+        sceneView.scene = makeScene()
+    
+        // Zoom scene view to the viewpoint specified by the camera position.
+        let point = AGSPoint(x: longitude, y: latitude, spatialReference: AGSSpatialReference.wgs84())
+        let camera = AGSCamera(lookAt: point, distance: 5500, heading: 150, pitch: 20, roll: 0)
+        sceneView.setViewpointCamera(camera)
+        
+        // Add graphics overlay to the scene
+        sceneView.graphicsOverlays.add(makeGraphicsOverlay())
+        
+        cameraControllers.append(contentsOf: [makeOrbitLocationCameraController(), makeOrbitGeoElementCameraController(), AGSGlobeCameraController()])
```

Comment:
Do you know how expensive camera controllers are to create and keep around? I'm just wondering if we could create the desired camera controller on demand based on the user's selection. Just a thought.